pkg/app/piped/controller/skipstage.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pipe-cd/pipecd/pkg/app/piped/executor"
+	"github.com/pipe-cd/pipecd/pkg/config"
+	"github.com/pipe-cd/pipecd/pkg/filematcher"
+	"github.com/pipe-cd/pipecd/pkg/git"
+)
+
+// checkSkipStage checks whether the stage should be skipped or not.
+func checkSkipStage(ctx context.Context, in executor.Input, opt config.SkipOptions) (skip bool, err error) {
+	if len(opt.Paths) == 0 && len(opt.CommitMessagePrefixes) == 0 {
+		// When no condition is specified.
+		return false, nil
+	}
```

Comment:
[imo] It might not be a func, just put it in the `shouldSkipStage` ?

Reply:
Thanks, I did it here [8a96a8c](https://github.com/pipe-cd/pipecd/pull/4922/commits/8a96a8c09fb01c1230275795fa3537e38574b7d7)